### PR TITLE
Implementing linear time suffix trimming

### PR DESCRIPTION
**Pull request by Yomguithereal**
Hello again @sindresorhus. This PR adds linear time suffix trimming to your function. This should give you an edge (at least it did on the benchmark I ran), but I advise you run your own just to be sure (maybe with more use cases and swapping the strings already present).

The idea is to trim the suffix of both strings linearly until they start to differ.

The strange `~-` operation used here is the bitwise way to perform a `-1` operation on a number. It speeds up the computation a bit on my benchmarks.

I can also implement prefix trimming but while writing this PR, I noticed that I did something wrong and my prefix trimming is actually erroneous so I need to fix it. However, since it means performing some more arithmetical operations, I fear it could slow down things rather than improving the performance of the function. I will come back later when my function is fixed to tell you if it's worth it to add prefix trimming then.

Have a nice day.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,6 +8,13 @@ module.exports = function (a, b) {
 		return 0;
 	}
 
+	var swap = a;
+
+	if (a.length > b.length) {
+		a = b;
+		b = swap;
+	}
```
Comment (sindresorhus) [inline index.js]: Can you add a short code comment on why you're swapping here?

File: index.js [hunk 2]
```diff
@@ -19,6 +26,15 @@ module.exports = function (a, b) {
 		return aLen;
 	}
 
+	while (aLen > 0 && (a.charCodeAt(~-aLen) === b.charCodeAt(~-bLen))) {
```
Comment (sindresorhus) [inline index.js]: Can you add a comment about what this does (suffix trimming)?

**Comment by Yomguithereal** [comment]
I've added some comments.

Note also that swapping `a` and `b` so `a` is the shortest could potentially boost the function even more if we decide to cache the charCodes of one or the other string but I need to think it a bit more.

**Comment by sindresorhus** [comment]
This is amazing. Thank you @Yomguithereal :)

**Comment by Yomguithereal** [comment]
I'll come back when I fix my prefix trimming (if this is still useful performance-wise) & if I find evidence that caching the longest or shortest or both char codes can be useful or not.

**Comment by sindresorhus** [comment]
Pretty good improvement actually: https://github.com/sindresorhus/leven/commit/cf30e86e35c60bc48dee910ca131e1327e856b87 🎉

**Pull request merged**